### ISSUE TITLE
Operetta missing metadata

### ISSUE THREAD

**Issue by tischi** [mobie/mobie-viewer-fiji#994]
@ThomasMicroscopy

The 5GB dataset called `Samples` that you shared with me has the following issue:

This file exists `r02c03f59p01-ch1sk1fk1fl1.tiff` in the folder but the `XML` document does not have it, but stops earlier:

<img width="343" alt="image" src="https://user-images.githubusercontent.com/2157566/224034351-13f00d8c-6ac7-43eb-8ab4-af41e5174489.png">

Do you know what could be going on?

From here it looks like as if this is a corrupt dataset...

What we maybe could do it to only accept files that are present in the XML.

**Comment by ThomasMicroscopy** [comment]
Indeed, also here on my computer these two files (2 channels) are missing in the samples folder,
in the original Image folder they are there.
However, I am looking presently at this Samples folder with no crash.

Maybe when creating the Samples folder, and trying to narrow down the files that caused the previous crash (the ones only about 40 kb in size)
I made a copy and paste error.

But since the MOBIE viewer can handle the missing 40kb image files, a normal missing image file shouldn’t be an issue?
At least here this folder is running right now fine.



On 9 Mar 2023, at 14:16, Christian Tischer ***@***.******@***.***>> wrote:



@ThomasMicroscopy<https://github.com/ThomasMicroscopy>

The 5GB dataset called Samples that you shared with me has the following issue:

This file exists r02c03f59p01-ch1sk1fk1fl1.tiff in the folder but the XML document does not have it, but stops earlier:

[image]<https://user-images.githubusercontent.com/2157566/224034351-13f00d8c-6ac7-43eb-8ab4-af41e5174489.png>

Do you know what could be going on?

From here it looks like as if this is a corrupt dataset...

What we maybe could do it to only accept files that are present in the XML.

**Comment by tischi** [comment]
> But since the MOBIE viewer can handle the missing 40kb image files, a normal missing image file shouldn’t be an issue?

It is not the image file that is missing but the **metadata is missing** in the XML.
How should we handle this?

(a) Crash or (b) ignore files that are not in the XML?

**Comment by ThomasMicroscopy** [comment]
Actually, these two entries which are missing in the xml are actually exactly those two small 40kb files.
So, obviously what happens, when a field is not recorded, because of a laser AF error, you get these two small cryptic 40kb TIFF files with no image content.  However, apparently in the .xml they are completely omitted. 
I had't not noticed so far that this happens.

**Comment by ThomasMicroscopy** [comment]
I guess it's good that these bad positions are left out in the .xml.   In the mobibigdataviewer I still have this open, and it just shows a black block, which is fine. So far no crash, so the missing entry in the xml is probably not an issue.

**Comment by tischi** [comment]
OK, this works now like this (and is not crashing anymore)

```
# MoBIE
Opening: /Volumes/cba/exchange/hcs-test/Samples
Files: 2930
Image data format: tiff
[WARNING] No metadata found for /Volumes/cba/exchange/hcs-test/Samples/r02c03f59p01-ch1sk1fk1fl1.tiff
[WARNING] No metadata found for /Volumes/cba/exchange/hcs-test/Samples/r02c03f59p01-ch2sk1fk1fl1.tiff
Initialised HCS plate: Samples
Images: 2926
Channels: 2
Wells: 24
Sites per well: 61
HCS Pattern: Operetta
````

Is that OK for you? Then I would close this issue.

**Comment by ThomasMicroscopy** [comment]
Yes, of course, if there is not metadata, then it looks like Harmony does not expect any viable tiff data there anyway
